packages/guides/src/Compiler/NodeTransformers/ClassNodeTransformer.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace phpDocumentor\Guides\Compiler\NodeTransformers;
+
+use phpDocumentor\Guides\Compiler\NodeTransformer;
+use phpDocumentor\Guides\Nodes\AnchorNode;
+use phpDocumentor\Guides\Nodes\ClassNode;
+use phpDocumentor\Guides\Nodes\DocumentNode;
+use phpDocumentor\Guides\Nodes\Node;
+
+/** @implements NodeTransformer<Node> */
+class ClassNodeTransformer implements NodeTransformer
+{
+
+    /**
+     * @var string[]
+     */
+    private array $classes = [];
+
+    public function enterNode(Node $node): Node
+    {
+        if ($node instanceof ClassNode) {
+            $this->classes = $node->getClasses();
```

Comment:
We do reset classes here, but should they not be reset when you start a new document?

Reply:
I tackled the issue of resetting and added tests for those scenarios